### ISSUE TITLE
Install Amplify via Carthage

### ISSUE THREAD

**Issue by lawmicha** [aws-amplify/amplify-ios#280]
- Install Amplify frameworks via carthage (https://github.com/Carthage/Carthage)
- Update README with instructions

**Comment by rodrigoelp** [comment]
I think you also need to include Swift package manager here... which was referenced here: #90

**Comment by lesmuc** [comment]
Any news on that? Would be really great to have support for Carthage or Swift Package Manager.

**Comment by rodrigoelp** [comment]
@lesmuc it would be amazing to have support for spm... but I highly doubt they will be prioritised any time soon.

**Comment by drochetti** [comment]
@lesmuc @rodrigoelp now that Amplify went GA, we will start looking at Swift Package Manager and Carthage.

**Comment by farshadtx** [comment]
@drochetti Is this new feature going to simplify the release?
[Distributing Binary Frameworks as Swift Packages](https://developer.apple.com/documentation/swift_packages/distributing_binary_frameworks_as_swift_packages)

**Comment by drochetti** [comment]
> @drochetti Is this new feature going to simplify the release?
> [Distributing Binary Frameworks as Swift Packages](https://developer.apple.com/documentation/swift_packages/distributing_binary_frameworks_as_swift_packages)

Possibly, we still need to look deeper into all SPM features to come up with our strategy for the support and release process.

**Comment by drochetti** [comment]
@farshadtx @rodrigoelp @lesmuc I created another issue to track SPM and Carthage separately: https://github.com/aws-amplify/amplify-ios/issues/578

**Comment by dmiluski** [comment]
For context, my company just moved off of CocoaPods as Firebase was breaking SwiftPreviews if codecov was enabled which was problematic for system feedback. Either Carthage or SwiftPM would be great to see.

https://github.com/CocoaPods/CocoaPods/issues/9275

So we moved off of cocoapods and are now saving 50% CI build time given cached builds.

**Comment by dmiluski** [comment]
Thinks for the follow up @drochetti ,

I ended up being able to unblock our work by utilizing the older aws-sdk-ios. Although I am a big fan of the publisher support, the aws-sdk-ios (which does support carthage) dependency helped us integrate with our current setup requirements.
github "aws-amplify/aws-sdk-ios" ~> 2.0

Reasons I've preferred Carthage:
Build Speed / Swift Preview Support

Mentioned above, (unrelated to AWS), Swift Previews were broken by Firebase and having quick feedback assisted in our development time. Additionally, with carthage caching, we were able to cut a good time chunk out of our CI jobs (20min to 5m). This does not mean carthage doesn't have pain points. We're currently stripping arm 64 architectures to get code working with Xcode 12, but cocoapods dependencies have also been going through pain with Xcode 12.
https://gist.github.com/skymobilebuilds/61f4a95bd62a3db36b52719aeab7e0d5

Ideally, our mid/long term goals are to move most of our dependencies to SwiftPM as they become available given the large company supported backing.

Given we have a work-a-round by using the older aws-sdk-ios, I am currently able to deliver on our goals even if Amplify doesn't introduce carthage support.
